src/main/deliferoo/view/Window.java

```diff
@@ -1,54 +1,56 @@
 package view;
 
 import java.io.File;
-import java.time.LocalTime;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import java.util.Random;
-
-import algorithm.Dijkstra;
-import algorithm.TSP1;
-import controller.Controller;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import javafx.geometry.Rectangle2D;
 import javafx.scene.Scene;
 import javafx.scene.control.Menu;
 import javafx.scene.control.MenuBar;
 import javafx.scene.control.MenuItem;
-import javafx.scene.control.TableView;
 import javafx.scene.layout.BorderPane;
-import javafx.scene.layout.GridPane;
 import javafx.scene.layout.VBox;
 import javafx.scene.paint.Color;
 import javafx.scene.shape.Rectangle;
 import javafx.stage.FileChooser;
 import javafx.stage.Screen;
 import javafx.stage.Stage;
+
+import algorithm.Dijkstra;
+import algorithm.TSP1;
+import controller.Controller;
 import model.BestPath;
 import model.Delivery;
 import model.Edge;
 import model.FullMap;
-import xml.XMLParser;
 
+/**
+ * Main window class
+ * 
+ * @author teck
+ * @author sadsitha
+ *
+ */
 public class Window {
 
     private Controller controller;
     private Rectangle2D bounds;
-    
-    FullMap map;
-    
     private MapView mapView;
     private TableBoxView tableBoxView;
-
-    private List<Delivery> deliveries;
+    private MessageView messageView;
+    private ObservableList<SpecialNodeTextView> specialNodeTextViews;
 
     public Window(Controller controller) {
 	this.controller = controller;
 	this.bounds = Screen.getPrimary().getVisualBounds();
 	this.mapView = new MapView(this.bounds.getHeight(), 2 * this.bounds.getWidth() / 3);
 	this.tableBoxView = new TableBoxView(this.bounds.getHeight() / 2, this.bounds.getWidth() / 3);
+	this.specialNodeTextViews = FXCollections.observableList(new ArrayList<SpecialNodeTextView>());
```

Comment:
not sure if we should make a variable for this because it is accessible from tableBoxView using .getItems()